wandb/sdk/launch/builder/templates/_wandb_bootstrap.py

```diff
@@ -67,7 +67,7 @@ def main() -> None:
                     # can't pip install wandb==0.*.*.dev1 through pip. Lets just install wandb for now
                     if req.startswith("wandb==") and "dev1" in req:
                         req = "wandb"
-                    reqs.append(req.strip())
+                    reqs.append(req.strip().replace(" ", ""))
```

Comment:
how are we instaling these deps later? feels weird to fix the whitespace versus fixing the eventual users of these deps. We may have other incompatibilities between this and standard pip so I wonder if there is a more systematic fix

Reply:
Yeah, in this specific case I think detecting `@` and just using the url after the `@` is the right thing to do.  There's got to be an official spec for `requirements.txt` somewhere.

Reply:
Like I imagine `pip install https://files.pythonhosted.org/packages/93/8d/e1e98360dc899e533cb3dd857494f2571b129bdffcee76365009b2bb507c/example_pypi_package-0.1.0.tar.gz` works fine, not sure about `pip install foo@https:...`

Reply:
I can move the change to `install_deps()` in the same file, and do the (likely more robust) fix of adding quotes around entries with spaces rather than removing all spaces.

There is an [official spec](https://pip.pypa.io/en/stable/reference/requirements-file-format/) for `requirements.txt`, but no easy `import ...` to interpret it that I could find, other than an [internal one that's not recommended](https://stackoverflow.com/a/49837302). That leaves the option of implementing the spec ourselves, which is a bit out of scope of this change IMO. For now we can just worry about the `"x @ y"` case.

Reply:
>do the (likely more robust) fix of adding quotes around entries with spaces rather than removing all spaces.

Couldn't quite manage this - the command kept ignoring the quotes even when they were escaped. I could get it to run by putting the command into a big string and using `shell=True`, but [that causes a security vulnerability](https://docs.python.org/3/library/subprocess.html#security-considerations).

Instead I took @vanpelt's suggestion of only using the url after the `@`, ignoring anything before it.